kolibri/core/lessons/serializers.py

```diff
@@ -128,7 +132,29 @@ def update(self, instance, validated_data):
                 lesson_id=instance.id,
                 collection_id__in=(set(current_group_ids) - set(new_group_ids))
             ).delete()
-
+        if validated_data.get("is_active", False):
+            #  When Lesson changes from inactive to active:
+            #  * All the summarylogs for the affected resources are reset
+            #  * All attemptlogs for the affected summarylogs are deleted
+            group_ids = set(instance.lesson_assignments.values_list('collection__id', flat=True))
+            user_ids = set()
+            for group_id in group_ids:
+                user_ids.update(
+                    set(
+                        HierarchyRelationsFilter(FacilityUser.objects.values_list('id', flat=True)).filter_by_hierarchy(
+                            target_user=F("id"), ancestor_collection=group_id
+                        )
+                    )
+                )
+            content_ids = [resource['content_id'] for resource in instance.resources]
+            summarylogs_ids = set(
+                ContentSummaryLog.objects.filter(content_id__in=content_ids)
+                .filter(user_id__in=user_ids)
+                .exclude(completion_timestamp__isnull=True)
+                .values_list('id', flat=True)
+            )
+            AttemptLog.objects.filter(masterylog__summarylog__in=summarylogs_ids).delete()
```

Comment:
Hrm, we do not want to delete the existing attemptlogs - missed this before.

This is why bumping to a new masterylog is important, because then we focus on the attemptlogs for this particular mastery log, without wiping out previous user engagement.